homeassistant/components/idasen_desk/__init__.py

```diff
@@ -15,62 +14,98 @@
     EVENT_HOMEASSISTANT_STOP,
     Platform,
 )
-from homeassistant.core import Event, HomeAssistant
-from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.core import Event, HomeAssistant, callback
 from homeassistant.helpers import device_registry as dr
 from homeassistant.helpers.device_registry import DeviceInfo
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
 from .const import DOMAIN
 
-PLATFORMS: list[Platform] = [Platform.COVER]
+PLATFORMS: list[Platform] = [Platform.COVER, Platform.SWITCH]
 
 _LOGGER = logging.getLogger(__name__)
 
 
+class IdasenDeskCoordinator(DataUpdateCoordinator):
```

Comment:
This PR is getting a bit large, I would move the coordinator changes to another PR (with it defaulted to on) and introduce the switch here with the ability to flip it from on to off

Reply:
Done: https://github.com/home-assistant/core/pull/102135